### PR TITLE
Check for pre-existing usernames on account creation

### PR DESCRIPTION
**Pull request by psybuzz**
This PR addresses the ability for multiple accounts with the same email.  In the first commit, a DB query is added to user.js that tries to find other accounts with the same email username.  If the number of results is greater than zero, it returns an error that is displayed to the user.

In the next commits, the validator module used for server-side validation is upgraded to a newer version that includes a "normalizeEmail" function.  This could be useful for normalizing different email addresses.  After this commit, for example, `amyiscool@gmail.com` will be seen as identical to `AmYisCOOL@gmail.com` after both strings are normalized.

### CODE REVIEW

**Comment by tonyxj** [comment]
A lot of files changed! 
LGTM and defaulting to @nikilselvam for a more in-depth look.

File: routes/user.js [hunk 1]
```diff
@@ -13,35 +13,42 @@ exports.create = function(req, res){
...
-		return resError(res, "Sorry, your name is too short or too long.  " +
-				"Do you have a nickname between 2 and 128 characters?");
+		return resError(res, "BAD_NAME_LENGTH");
 	}
 
-	// Hash the password.
-	var passHash = bcrypt.hashSync(req.body.password, salt);
+	// Check for another pre-existing user with the same email.
+	User.find({ username: validator.escape(email) }, 'username', function(err, docs){
+		// Stop the request if there was a DB error of it another user has the same username.
```
Comment (nikilselvam) [inline routes/user.js]: I think this comment may have a grammatical error in it. Is the `of it` part correct in the middle or is it meant to be something else?

**Comment by nikilselvam** [comment]
Great! There's one comment that I think needs to be reworded a little bit for clarity, but other than that this LGTM.

File: routes/user.js (hunk 1 above)
Comment (psybuzz) [inline routes/user.js]: Ahh, you're right, it was supposed to be "...stop if there was an error OR if another user has..."
Thanks Nikil!

**Pull request merged**